solidity/test/rewards/TestECDSARewardsDistributorEscrow.js

```diff
@@ -0,0 +1,133 @@
+const {accounts, contract, web3} = require("@openzeppelin/test-environment")
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+const {expectRevert} = require("@openzeppelin/test-helpers")
+const {assert, expect} = require("chai")
+
+const KeepToken = contract.fromArtifact("KeepToken")
+const PhasedEscrow = contract.fromArtifact("PhasedEscrow")
+const ECDSARewardsEscrowBeneficiary = contract.fromArtifact(
+  "ECDSARewardsEscrowBeneficiary"
+)
+const ECDSARewardsDistributor = contract.fromArtifact("ECDSARewardsDistributor")
+const ECDSARewardsDistributorEscrow = contract.fromArtifact(
+  "ECDSARewardsDistributorEscrow"
+)
+
+describe("ECDSARewardsDistributorEscrow", () => {
+  const owner = accounts[0]
+  const thirdParty = accounts[1]
+
+  let token
+  let rewardsDistributor
+  let escrow
+
+  before(async () => {
+    token = await KeepToken.new({from: owner})
+    rewardsDistributor = await ECDSARewardsDistributor.new(token.address, {
+      from: owner,
+    })
+    escrow = await ECDSARewardsDistributorEscrow.new(
+      token.address,
+      rewardsDistributor.address,
+      {from: owner}
+    )
+
+    await rewardsDistributor.transferOwnership(escrow.address, {from: owner})
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("allocateInterval", async () => {
+    const merkleRoot =
+      "0x65b315f4565a40f738cbaaef7dbab4ddefa14620407507d0f2d5cdbd1d8063f6"
+    const amount = web3.utils.toBN(999998997)
+
+    beforeEach(async () => {
+      const allTokens = await token.balanceOf(owner)
+      await token.approveAndCall(escrow.address, allTokens, "0x0", {
+        from: owner,
+      })
+    })
+
+    it("can not be called by non-owner", async () => {
+      await expectRevert(
+        escrow.allocateInterval(merkleRoot, amount, {from: thirdParty}),
+        "Ownable: caller is not the owner"
+      )
+    })
+
+    it("can be called by owner", async () => {
+      await escrow.allocateInterval(merkleRoot, amount, {from: owner})
+      // ok, no reverts
+    })
+
+    it("allocates reward distribution", async () => {
+      await escrow.allocateInterval(merkleRoot, amount, {from: owner})
+
+      const eventList = await rewardsDistributor.getPastEvents(
+        "RewardsAllocated",
+        {
+          fromBlock: 0,
+          toBlock: "latest",
+        }
+      )
+
+      assert.equal(eventList.length, 1, "incorrect number of emitted events")
```

Comment:
Do you think we can use `expect` here and in all the rest assertions? We mix 2 approaches in this test (assert/expect), but it looks like we can use `expect` only.

`expect(eventList.length).to.equal(1, "incorrect number of emitted events")`

Reply:
Good point, I'll fix it!

Reply:
https://github.com/keep-network/keep-ecdsa/pull/631/commits/a37f774f429f1c8c431d9404c87cb809489b95ff